### PR TITLE
[Colab-2755] Judge Statistics for Contest View

### PR DESCRIPTION
**Pull request by cwoebker**
First draft for the judge statistics. I am still testing and fixing some last issues.

---
This change is [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)

### CODE REVIEW

**Comment by jobachhu** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 183 at r1](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCQdJdT3iB1F-OSliza:-LCQdJdT3iB1F-OSlizb:b5zdoj0) ([raw file](https://github.com/cci-mit/xcolab/blob/292bd3382a0949c14b47aa6ecc7d3a16c40fa3f5/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L183)):*
> ```
>                 <div class="p-ContestProposals__header__team">
>                     <h2>Admin Information</h2>
>                     <c:if test="${showEditLink}">
> ```

I'd maybe give this different permission name. Does it make sense for this to be the same as the export button? I think these might only be used by the same type of people, right? My thinking in keeping them separate from the edit button as that those might be performed by different roles in the future (it makes sense for someone to edit a contest but not to be involved in its judging). I can't think of a scenario where someone would want to download the CSV and not see judge status (or vice versa) - what do you think?

My two thoughts:
1. only tie it to fellow permission
2. create a `getCanAdminJudging()` method or something similar

---


*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)*

File: view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx [hunk 1]
```diff
@@ -175,6 +177,37 @@
                         </a>
                     </c:if>
                 </div>
+
+                <c:if test="${showJudgingStatus}">
```
Comment (cwoebker) [inline view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx]: @jobachhu this one was already available from contestProposals.jspx, i guess it is best to reuse in this case. Missed this one earlier.

**Comment by cwoebker** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 194 at r2](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCVdZvoABjvGDXz-dEA:-LCVdZvpBzzpIFGpXbtP:b6509wf) ([raw file](https://github.com/cci-mit/xcolab/blob/cc6f1bd2596eed465827b0ea1f1d19b097ce9714/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L194)):*
> ```
>                                         </div>
>                                         <br/>
>                                         <judging:judgeAssignedProposalCount userId="${judge.userId}" contestPhaseId="${contestPhase.contestPhasePK}">
> ```

Unfortunately the judgeAssignedProposalCount doesn't work for me here. userId and contestPhaseId should be set correctly. Are there any other pitfalls to be aware of?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)*

**Comment by jobachhu** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 181 at r2](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCPl_1M5zr3VjiMxAE3-r2-181:-LCVtdsvEy9kpgZ7oFp5:b-ovwawk) ([raw file](https://github.com/cci-mit/xcolab/blob/cc6f1bd2596eed465827b0ea1f1d19b097ce9714/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L181)):*
<details><summary><i>Previously, cwoebker (Cecil Wöbker) wrote…</i></summary><blockquote>

@jobachhu this one was already available from contestProposals.jspx, i guess it is best to reuse in this case. Missed this one earlier.
</blockquote></details>

Good idea!

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 194 at r2](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCVdZvoABjvGDXz-dEA:-LCVvSa98Eo0zQM9wCSK:b1jg0n8) ([raw file](https://github.com/cci-mit/xcolab/blob/cc6f1bd2596eed465827b0ea1f1d19b097ce9714/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L194)):*
<details><summary><i>Previously, cwoebker (Cecil Wöbker) wrote…</i></summary><blockquote>

Unfortunately the judgeAssignedProposalCount doesn't work for me here. userId and contestPhaseId should be set correctly. Are there any other pitfalls to be aware of?
</blockquote></details>

Hm, no pitfalls that I'm aware of - the variables should be set. Can you set a breakpoint in `org.xcolab.view.pages.proposals.judging.JudgeAssignedProposalCountTag#doStartTag` to see if the input values are received and the output is set in `pageContext` correctly? 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)*

**Comment by cwoebker** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 183 at r1](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCQdJdT3iB1F-OSliza:-LCk9ugH1NJUhMR9dSWD:b-896fix) ([raw file](https://github.com/cci-mit/xcolab/blob/292bd3382a0949c14b47aa6ecc7d3a16c40fa3f5/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L183)):*
<details><summary><i>Previously, jobachhu (Johannes Bachhuber) wrote…</i></summary><blockquote>

I'd maybe give this different permission name. Does it make sense for this to be the same as the export button? I think these might only be used by the same type of people, right? My thinking in keeping them separate from the edit button as that those might be performed by different roles in the future (it makes sense for someone to edit a contest but not to be involved in its judging). I can't think of a scenario where someone would want to download the CSV and not see judge status (or vice versa) - what do you think?

My two thoughts:
1. only tie it to fellow permission
2. create a `getCanAdminJudging()` method or something similar
</blockquote></details>

Done.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 194 at r2](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCVdZvoABjvGDXz-dEA:-LCk9mGf4CD28AZBSfn5:b-gb4qm9) ([raw file](https://github.com/cci-mit/xcolab/blob/cc6f1bd2596eed465827b0ea1f1d19b097ce9714/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L194)):*
<details><summary><i>Previously, jobachhu (Johannes Bachhuber) wrote…</i></summary><blockquote>

Hm, no pitfalls that I'm aware of - the variables should be set. Can you set a breakpoint in `org.xcolab.view.pages.proposals.judging.JudgeAssignedProposalCountTag#doStartTag` to see if the input values are received and the output is set in `pageContext` correctly? 
</blockquote></details>

They are being passed through. But both are zero. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)*

**Comment by jobachhu** [comment]
Reviewed 2 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx, line 194 at r2](https://reviewable.io/reviews/cci-mit/xcolab/132#-LCVdZvoABjvGDXz-dEA:-LD6RSTF6xT4jCABMUrI:boeepp1) ([raw file](https://github.com/cci-mit/xcolab/blob/cc6f1bd2596eed465827b0ea1f1d19b097ce9714/view/src/main/webapp/WEB-INF/jsp/proposals/contestProposals/header_contest_details.jspx#L194)):*
<details><summary><i>Previously, cwoebker (Cecil Wöbker) wrote…</i></summary><blockquote>

They are being passed through. But both are zero. 
</blockquote></details>

Hm this is strange. Have you tried printing them before they get passed into the tag? Are they also zero there? Does the tag work elsewhere?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132)*

**Comment by jobachhu** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/132#-:-LDH4H019Ck42N783EpY:bnfp4nl)*

**Comment by jobachhu** [comment]
Thanks Cecil!

**Pull request merged**